File: integerToEnglishWords.cpp
class Solution {
public:
 string below20[20] ={"","One","Two","Three","Four","Five","Six","Seven","Eight","Nine","Ten","Eleven","Twelve","Thirteen","Fourteen","Fifteen","Sixteen","Seventeen","Eighteen","Nineteen"};
    
    string tens[10]={"","","Twenty","Thirty","Forty","Fifty","Sixty","Seventy","Eighty","Ninety"};
    string thousands[4]={"","Thousand","Million", "Billion"};
    
    string helper(int num) {
        if (num==0) {
          return "";  
        } else if (num<20) {
           return below20[num]+" ";
        } else if (num<100) {
            return tens[num/10]+" "+ helper(num%10);//below20[num%10];   
        } else {
            return below20[num/100]+" Hundred "+ helper(num%100);
        }
    }
    string numberToWords(int num) {
        if (num==0)
            return "Zero";
        
        int idx=0;
        string answer;
        while (num>0) {
            if (num%1000!=0) {
                answer=helper(num%1000)+thousands[idx]+" "+answer;
            }
            num=num/1000;
            idx++;
        }
        while (answer[answer.length()-1]==' ')
        answer.erase(answer.begin()+answer.length()-1);
        return answer;
    }
};

File: integer_to_English_Words.py
"""
Time Complexity: O(16) = O(1) Billio can have maximum 4 triplets, to convert each tripet to word maximum 4 operations are required
Space Complexity: O(1) 

Did this code successfully run on Leetcode : Yes
Any problem you faced while coding this : No

"""


class Solution:
    def __init__(self):
        self.below_20 = ["", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten",
                         "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen"]
        self.tens = ["", "Ten", "Twenty", "Thirty", "Forty",
                     "Fifty", "Sixty", "Seventy", "Eighty", "Ninety"]
        self.thousands = ["", "Thousand", "Million", "Billion"]

    def numberToWords(self, num: int) -> str:
        if num == 0:
            return "Zero"
        i = 0
        result = ""
        while num > 0:
            if num % 1000 != 0:
                result = self.helper(num % 1000) + \
                    self.thousands[i] + " " + result

            num = num // 1000
            i += 1

        return result.strip()

    def helper(self, num):
        if num == 0:
            return ""
        if num < 20:
            return self.below_20[num] + " "
        if num < 100:
            return self.tens[int(num/10)] + " " + self.helper(num % 10)
        if num < 1000:
            return self.below_20[int(num/100)] + " Hundred " + self.helper(num % 100)
